.script/tests/detectionTemplateSchemaValidation/DetectionsYamlFilesTestData.cs

```diff
@@ -67,10 +67,45 @@ private static string GetAssemblyDirectory()
 
         private static List<string> GetDetectionFiles(List<string> detectionPaths)
         {
-            var files = Directory.GetFiles(detectionPaths[0], "*.yaml", SearchOption.AllDirectories).ToList();
-            files.AddRange(Directory.GetFiles(detectionPaths[1], "*.yaml", SearchOption.AllDirectories).ToList().Where(s => s.Contains("Analytic Rules")));
+            int prNumber = 0;
+            int.TryParse(Environment.GetEnvironmentVariable("PRNUM"), out prNumber);
+            //assign pr number to debug with a pr
+            //prNumber=8414;
+            var files = Directory.GetFiles(detectionPaths[0], "*.yaml", SearchOption.AllDirectories)
+                .Concat(Directory.GetFiles(detectionPaths[1], "*.yaml", SearchOption.AllDirectories)
+                .Where(s => s.Contains("Analytic Rules")));
 
-            return files;
+            if (prNumber != 0)
+            {
+                try
+                {
+                    var client = new GitHubClient(new ProductHeaderValue("MicrosoftSentinelValidationApp"));
+                    var prFiles = client.PullRequest.Files("Azure", "Azure-Sentinel", prNumber).Result;
+                    var prFilesListModified = new List<string>();
+                    var basePath = GetRootPath();
+                    foreach (var file in prFiles)
+                    {
+                        var modifiedFile = Path.Combine(basePath, file.FileName.Replace('/', Path.DirectorySeparatorChar));
+                        prFilesListModified.Add(modifiedFile);
+                    }
+
+                    files = files.Where(file => prFilesListModified.Any(prFile => file.Contains(prFile)));
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Error occured while getting the files from PR. Error message: " + ex.Message + " Stack trace: " + ex.StackTrace);
+                }
+            }
+
+            var fileList = files.ToList();
```

Comment:
Can you please test in any case can we have "files =null"? e.g. in case of prNumber =0? If so, we can change above if condition to if prNumber = 0 then add file and return.